sdk/core/azure-core/inc/azure/core/http/websockets/curl_websockets_transport.hpp

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief #Azure::Core::Http::WebSockets::WebSocketTransport implementation via CURL.
+ */
+
+#pragma once
+
+#include "azure/core/context.hpp"
+#include "azure/core/http/curl_transport.hpp"
+#include "azure/core/http/http.hpp"
+#include "azure/core/http/transport.hpp"
+#include "azure/core/http/websockets/websockets_transport.hpp"
+#include <memory>
+
+namespace Azure { namespace Core { namespace Http { namespace WebSockets {
+
+  struct CurlWebSocketTransportOptions : public Azure::Core::Http::CurlTransportOptions
+  {
+  };
+  /**
+   * @brief Concrete implementation of a WebSocket Transport that uses libcurl.
+   */
+  class CurlWebSocketTransport : public CurlTransport, public WebSocketTransport {
+  public:
+    /**
+     * @brief Construct a new CurlWebSocketTransport object.
+     *
+     * @param options Optional parameter to override the default options.
+     */
+    CurlWebSocketTransport(
+        CurlWebSocketTransportOptions const& options = CurlWebSocketTransportOptions())
+        : CurlTransport(options)
+    {
+    }
+
+    /**
+     * @brief Implements interface to send an HTTP Request and produce an HTTP RawResponse
+     *
+     * @param request an HTTP Request to be send.
+     * @param context A context to control the request lifetime.
+     *
+     * @return unique ptr to an HTTP RawResponse.
+     */
+    virtual std::unique_ptr<RawResponse> Send(Request& request, Context const& context) override;
+
+    /**
+     * @brief Indicates if the transport natively supports websockets or not.
+     *
+     * @details For the CURL websocket transport, the transport does NOT support native websockets -
+     * it is the responsibility of the client of the WebSocketTransport to format WebSocket protocol
+     * elements.
+     */
+    virtual bool HasBuiltInWebSocketSupport() override { return false; }
+
+    /**
+     * @brief Closes the WebSocket handle.
```

Comment:
close not closes, in memory of Victor

Reply:
Umm. I disagree. My rule on method declarations is that the help text should say what the function does. In this case, you would never say "this function close the websocket", but you would say "this function closes the websocket".

The *caller* of this function wants to "close the WebSocket handle", but the function closes the websocket.